bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/handler/NuvoHandler.java

```diff
@@ -1135,21 +1133,18 @@ private void enableNuvonet(boolean showReady) {
                                 sourceMenuStateOptions);
                     }
 
-                    String[] favorites = favoriteMap.get(source);
-                    if (favorites != null) {
-                        connector.sendCommand(source.getId() + "FAVORITES"
-                                + (favorites.length < 20 ? favorites.length : 20) + COMMA
-                                + (source.getNum() == 1 ? ONE : ZERO) + COMMA + (source.getNum() == 2 ? ONE : ZERO)
-                                + COMMA + (source.getNum() == 3 ? ONE : ZERO) + COMMA
-                                + (source.getNum() == 4 ? ONE : ZERO) + COMMA + (source.getNum() == 5 ? ONE : ZERO)
-                                + COMMA + (source.getNum() == 6 ? ONE : ZERO));
-                        Thread.sleep(SLEEP_BETWEEN_CMD_MS);
+                    final String[] favorites = favoriteMap.getOrDefault(source, new String[0]);
+                    connector.sendCommand(source.getId() + "FAVORITES" + (favorites.length < 20 ? favorites.length : 20)
```

Comment:
Since this is a change in behavior, I'm wonder if it is intended to send a command when no favorite is found for source?

Reply:
Thanks for the explanation, makes sense.